thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js

```diff
@@ -51,7 +159,68 @@ export default Ember.Component.extend({
         const newContext = { urns, anomalyRange, baselineRange, analysisRange };
         onChange(newContext);
       }
+    },
+
+    /**
+     * Sets the new anomaly region date in ms
+     * @method setAnomalyDateRange
+     * @param {String} start  - stringified start date
+     * @param {String} end    - stringified end date
+     * @return {undefined}
+     */
+    setAnomalyDateRange(start, end) {
+      const anomalyRangeStart = moment(start).valueOf();
+      const anomalyRangeEnd = moment(end).valueOf();
+
+      this.setProperties({
+        anomalyRangeStart,
+        anomalyRangeEnd
+      });
+    },
+    
+    /**
+     * Sets the new display date in ms
+     * @method setDisplayDateRange
+     * @param {String} start  - stringified start date
+     * @param {String} end    - stringified end date
+     * @return {undefined}
+     */
+    setDisplayDateRange(start, end) {
+      const analysisRangeStart = moment(start).valueOf();
+      const analysisRangeEnd = moment(end).valueOf();
+
+      this.setProperties({
+        analysisRangeStart,
+        analysisRangeEnd
+      });
+    },
+
+    /**
+     * Changes the compare mode
+     * @method onModeChange
+     * @param {String} compareMode baseline compare mode
+     * @return {undefined}
+     */
+    onModeChange(compareMode) {
+      const {
+        analysisRangeStart,
+        analysisRangeEnd
+      } = this.getProperties('analysisRangeStart', 'analysisRangeEnd');
+      const offset = {
```

Comment:
maybe define this as a constant further above. also, it seems this subtracts N days, rather than weeks

Reply:
good catch, meant to be weeks 👍